core/src/data_sources/data_source.rs

```diff
@@ -730,25 +734,144 @@ impl DataSource {
+                            d.chunks = chunks;
+                            return Ok(d);
+                        }
+                        // this just adds the same amount per chunk, probably would want to add as much as possible,
+                        // curious about thoughts
+                        // TODO: also make it add below or above rather than not at all if not enough space
+                        let num_close_addable =
+                            (context_length - current_length) / (chunk_size * chunks.len() * 2);
+                        for chunk in chunks.iter() {
+                            // iterate through min to offset:
+                            for i in cmp::min(chunk.offset as i64 - num_close_addable as i64, 0)
+                                as usize
+                                ..cmp::max(chunk.offset + num_close_addable, d.chunk_count) + 1
+                            {
+                                if !offset_set.contains(&i) {
+                                    cur_chunks.push(i as i64);
+                                    offset_set.insert(i);
+                                }
+                            }
+                        }
+                        let filter = qdrant::Filter {
+                            must: vec![
+                                qdrant::FieldCondition {
+                                    key: "document_id".to_string(),
+                                    r#match: Some(qdrant::Match {
+                                        match_value: Some(qdrant::r#match::MatchValue::Keyword(
+                                            d.document_id.clone(),
+                                        )),
+                                    }),
+                                    ..Default::default()
+                                }
+                                .into(),
+                                qdrant::FieldCondition {
+                                    key: "chunk_offset".to_string(),
+                                    r#match: Some(qdrant::Match {
+                                        match_value: Some(qdrant::r#match::MatchValue::Integers(
+                                            qdrant::RepeatedIntegers {
+                                                integers: cur_chunks,
+                                            },
+                                        )),
+                                    }),
+                                    ..Default::default()
+                                }
+                                .into(),
+                            ],
+                            ..Default::default()
+                        };
+                        // use search_points
+                        let search_points = qdrant::ScrollPoints {
+                            collection_name: collection,
+                            filter: Some(filter),
+                            limit: Some(1000),
```

Comment:
What should I do here? Set the limit to the number of chunk we want to retrieve?

Reply:
Yes def!